### PR TITLE
added GSM_code

### PR DESCRIPTION
**Pull request by jaya-shankar**
GSM code that can handle power cuts, by restarting or whenever there is failure at sim card side

### CODE REVIEW

File: GSM_code/GSM_code.ino [hunk 1]
```diff
@@ -25,46 +27,64 @@ void setup()
...
+
+
+
+//   ----- CHECKING ACTION STATE ---------
+     if(actionState == AS_LOST_CONNECTION){
+      Serial.println("Some Error Occured , Resetting");
+      try
+        wait = wait*wait;
+      catch{}
+      setup();
```
Comment (rkpagadala) [inline GSM_code/GSM_code.ino:73]: call set_up_GSM

**Comment by rkpagadala** [comment]
@jaya-shankar 
This branch has conflicts that must be resolved
 to resolve conflicts before continuing.
Conflicting files
GSM_code/connections.md

**Comment by jaya-shankar** [comment]
The conflict is because only 
I changed the connections file to .txt from .md
that's it & there is no change in code
Should I delete the txt file?

**Comment by rkpagadala** [comment]
do what's needed

**Comment by jaya-shankar** [comment]
okay

File: GSM_code/helpers.h [hunk 2]
```diff
@@ -156,10 +156,10 @@ void parseATText(byte b) {
       if ( contentLength <= 0 ) {
 
         // all content bytes have now been read
-        add_to_response("status_code",status_code);
+        add_to_response("sc",status_code);
         parseState = PS_DETECT_MSG_TYPE;
         resetBuffer();
-        response.toCharArray(response_arr, response.length());
+//        response.toCharArray(response_arr, response.length());
```
Comment (rkpagadala) [inline GSM_code/helpers.h:162]: not needed?

File: GSM_code/GSM_code.ino [hunk 3]
```diff
@@ -3,50 +3,62 @@
...
 
 SoftwareSerial GSM(recevier, transferer); // RX-pin7, TX-pin8
+  
 #include "helpers.h"    // GSM cmds parser
 #include "requests.h"   // Functions for GET & POST
 #include "hcsr04.h"     // HCSR04 sensor code
 #include "relay.h"      // Relay Switch code
 
-String APN_name = "airtelgprs.com";               // change the APN name as per the SIM card
+String APN_name = "bsnlnet";                            // change the APN name as per the SIM card
```
Comment (rkpagadala) [inline GSM_code/GSM_code.ino:12]: how do we know this?
leave the old one commented out

**Pull request merged**